src/datastore/postgres-store.ts

```diff
@@ -3754,6 +3998,26 @@ export class PgDataStore
 
   cachedParameterizedInsertStrings = new Map<string, string>();
 
+  generateParameterizedWhereAndOrClause(
```

Comment:
The query output of this function is hard to reason about, and I'm not sure it's correct.
It looks like it could output statements like `WHERE ... AND ... OR ... AND ... ` which don't have parenthesis around them so unclear about order of operations, e.g. will it run `WHERE (... AND ...) OR (... AND ...)` or `WHERE ... AND ( ... OR ... AND ...)`, etc. It looks like only one function is calling this, can that function be simplified for a more readable query?

Reply:
The requirement was to gather results based on two attributes (`tx_id` and `index_block_hash`) within events.
You're right. I'll change the query to `WHERE ( ... AND ... ) OR ( ... AND ... )` as we want results with similar `tx_id` and `index_block_hash`.
Regarding the readability, do you think loop is more preferred than recursion?